diff --git a/dist/BunHttpServer.js b/dist/BunHttpServer.js
index 752913a..8ef0c5a 100644
--- a/dist/BunHttpServer.js
+++ b/dist/BunHttpServer.js
@@ -45,6 +45,10 @@ export const make = /*#__PURE__*/Effect.fnUntraced(function* (options) {
     ...options,
     fetch: handlerStack[0],
     websocket: {
+      perMessageDeflate: {
+        compress: "dedicated",
+        decompress: "shared"
+      },
       open(ws) {
         Deferred.doneUnsafe(ws.data.deferred, Exit.succeed(ws));
       },
diff --git a/src/BunHttpServer.ts b/src/BunHttpServer.ts
index 6a6c67b..1160090 100644
--- a/src/BunHttpServer.ts
+++ b/src/BunHttpServer.ts
@@ -111,6 +111,10 @@ export const make = Effect.fnUntraced(
       ...options as ServeOptions<R>,
       fetch: handlerStack[0],
       websocket: {
+        perMessageDeflate: {
+          compress: "dedicated",
+          decompress: "shared"
+        },
         open(ws) {
           Deferred.doneUnsafe(ws.data.deferred, Exit.succeed(ws))
         },
